enterprise/models.py

```diff
@@ -321,6 +368,147 @@ def __repr__(self):
         return self.__str__()
 
 
+class UserDataSharingConsentAudit(TimeStampedModel):
+    """
+    Object that exists to store the canonical state of whether a particular
+    user has given consent for their course data to be shared with a particular
+    single-sign-on identity provider
+    """
+    class Meta(object):
+        app_label = 'enterprise'
+        verbose_name = "Data Sharing Consent Audit State"
+        verbose_name_plural = "Data Sharing Consent Audit States"
+        unique_together = ('user', 'enterprise_customer',)
+
+    NOT_SET = 'not_set'
+    ENABLED = 'enabled'
+    DISABLED = 'disabled'
+    STATE_CHOICES = (
+        (NOT_SET, 'Not set'),
+        (ENABLED, 'Enabled'),
+        (DISABLED, 'Disabled'),
+    )
+
+    user = models.ForeignKey(
+        User,
+        on_delete=models.CASCADE,
+    )
+
+    enterprise_customer = models.ForeignKey(
+        EnterpriseCustomer,
+        on_delete=models.CASCADE,
+    )
```

Comment:
@bradenmacdonald, I think this is a point where we need some clarification from @mattdrayer. My understanding was that the end goal would be that logging in with an identity provider associated to an EnterpriseCustomer would associate the user in question with the EnterpriseCustomer as well, without requiring the manual process of adding the user first.

To get this working without the creation of an EnterpriseCustomer is complicated - for example, we don't have a user email until after the registration form on which data sharing consent is requested is rendered. And, we could end up with a situation where the user has actually given consent (on that registration form), but we have no way of recording it.

If the goal is actually to require manual entry of each EnterpriseCustomerUser, then that's one thing, but I don't think it is.